### DISCUSSION TITLE
Hash Support

### DISCUSSION THREAD

**Discussion by garysm** [ra1u/redis-dart#71] [Q&A] [answered]
Is there any support for hashes ([https://redis.io/docs/data-types/hashes/](url))?

If so, what would that look like in a Command?

**Comment by ra1u** [comment, marked as answer]
You can start with something like

~~~ dart
Command cmd = await RedisConnection().connect('localhost', 6379);
var i = await cmd.send_object(["HSET", "user:123", "username", "martina", "firstName", "Martina", "lastName", "Elisa", "country","GB"]);
print("hash size is now $i");
var user = await cmd.send_object(["HGET", "user:123", "username");
print("user $user");
var all = await cmd.send_object("HGETALL", "user:123");
print("user $all");
~~~

I don't have compiler here and for this reason you will likely need to do some tweaking. Let me know if you have difficulties making this example to run. Fell free to post your suggestions, ideas or questions on how we can move forward from here.

Kind regards, Luka

**Reply by garysm** [reply]
Thank you for the response @ra1u, I will work with this and get back to you.

**Reply by MilindModi** [reply]
I tried this solution it's working but adding value as key in the same set. I've attached Screenshots and code for better understanding.
Also tried adding multiple keys and multiple values in one statement but it's resulting in the same.
Code :
```
  Future postChatList(String data) async {
    await res.send_object(["HSET",user_id,data,"XYZ"]);
  }
```
<img width="504" alt="Screenshot 2023-02-08 at 2 14 58 PM" src="https://user-images.githubusercontent.com/28483876/217480244-71dbc1d2-4a1f-4333-b912-5cf2b8d142e7.png">

<img width="500" alt="Screenshot 2023-02-08 at 2 20 42 PM" src="https://user-images.githubusercontent.com/28483876/217480766-d49e87ce-79d6-4600-9105-65cdd2108231.png">

**Reply by ra1u** [reply]
@MilindModi I am sorry I do not understand what you problem is. Can you please explain following.

1. What you are trying to do 
2. What are you expecting to happen
3. What are you doing
4. What result you get that is different from your expectations

Kind regards, Luka

**Reply by MilindModi** [reply]
My apologize for not thoroughly researching the subject matter before commenting.  i was not aware about the basic concept that HSET also sets the value under the same key, 
Like i tried directly running  this `hset main_user_id user_id1 value1 user_id2 value2` directly on terminal  `hgetall main_user_id`
at that time i came to knew that value of sub field's value also gets stored in the same set. 
In nutshell provided answer by @ra1u  is correct.
